File: app-miyou/src/main/java/com/miyou/app/application/dialogue/policy/SttPolicy.java
package com.miyou.app.application.dialogue.policy;

/** STT 처리 정책 — 파일 크기 제한 및 기본 언어를 보유합니다. */
public record SttPolicy(
	long maxFileSizeBytes,
	String defaultLanguage
) {
	public SttPolicy {
		if (maxFileSizeBytes <= 0) {
			throw new IllegalArgumentException("maxFileSizeBytes must be positive");
		}
		if (defaultLanguage == null || defaultLanguage.isBlank()) {
			throw new IllegalArgumentException("defaultLanguage must not be blank");
		}
		defaultLanguage = defaultLanguage.trim();
	}
}

File: backend/application/src/main/kotlin/com/miyou/app/application/dialogue/policy/SttPolicy.kt
package com.miyou.app.application.dialogue.policy

data class SttPolicy(
    val maxFileSizeBytes: Long,
    private val rawDefaultLanguage: String,
) {
    val defaultLanguage: String = rawDefaultLanguage.trim()

    init {
        require(maxFileSizeBytes > 0) { "maxFileSizeBytes must be positive" }
        require(this.defaultLanguage.isNotBlank()) { "defaultLanguage must not be blank" }
    }
}
